src/m_menu.c

```diff
@@ -797,12 +800,78 @@ menu_t LoadDef =
   &MainDef,
   LoadMenu,
   M_DrawLoad,
-  80,34, //jff 3/15/98 move menu up
+  M_X_LOADSAVE,34, //jff 3/15/98 move menu up
   0
 };
 
 #define LOADGRAPHIC_Y 8
 
+// [FG] draw a snapshot of the n'th savegame into a separate window,
+//      or fill window with solid color and "n/a" if snapshot not available
+
+static int snapshot_width, snapshot_height;
+
+static void M_DrawBorderedSnapshot (int n)
+{
+  int x, y;
+  patch_t *patch;
+  char *txt;
+
+  const int snapshot_x = MAX((WIDESCREENDELTA + SaveDef.x + SKULLXOFF - snapshot_width) / 2, 8);
+  const int snapshot_y = LoadDef.y + MAX((load_end * LINEHEIGHT - snapshot_height) * n / load_end, 0);
+
+  // [FG] a snapshot window smaller than 80*48 px is considered too small
+  if (snapshot_width < ORIGWIDTH/4)
+    return;
+
+  if (!M_DrawSnapshot(n, snapshot_x, snapshot_y, snapshot_width, snapshot_height))
+  {
+    txt = "n/a";
+    M_WriteText(snapshot_x + snapshot_width/2 - M_StringWidth(txt)/2 - WIDESCREENDELTA,
+                snapshot_y + snapshot_height/2 - M_StringHeight(txt)/2,
+                txt);
+  }
+
+  txt = M_GetSavegameTime(n);
+  M_DrawString(snapshot_x + snapshot_width/2 - M_GetPixelWidth(txt)/2 - WIDESCREENDELTA,
+               snapshot_y + snapshot_height + M_StringHeight(txt),
+               CR_GOLD, txt);
+
+  // [FG] draw the view border around the snapshot
```

Comment:
I see that this code is taken from `R_FillBackScreen()`. How about a new function (eg `R_DrawBorder()`) to reduce code duplication?

Reply:
> One little nitpick - most other ports don't draw STBAR/ui in snapshots. Have you thought about it?

No, I haven't thought about it.

> I see that this code is taken from `R_FillBackScreen()`. How about a new function (eg `R_DrawBorder()`) to reduce code duplication?

Very good idea, will work on it post-PR.